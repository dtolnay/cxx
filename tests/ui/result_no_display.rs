#[cxx::bridge]
mod ffi {
    extern "Rust" {
        fn f() -> Result<()>;
    }
}

pub struct NonError;

fn f() -> Result<(), NonError> {
    Ok(())
}

fn main() {}
