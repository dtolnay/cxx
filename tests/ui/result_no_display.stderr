error[E0599]: the method `to_cxx_exception` exists for reference `&NonError`, but its trait bounds were not satisfied
 --> tests/ui/result_no_display.rs:4:19
  |
4 |         fn f() -> Result<()>;
  |                   ^^^^^^^^^^ method cannot be called on `&NonError` due to unsatisfied trait bounds
...
8 | pub struct NonError;
  | ------------------- doesn't satisfy `NonError: std::fmt::Display`
  |
  = note: the following trait bounds were not satisfied:
          `NonError: std::fmt::Display`
          which is required by `&NonError: ToCxxExceptionDefault`
note: the trait `std::fmt::Display` must be implemented
 --> $RUST/core/src/fmt/mod.rs
  = note: this error originates in the macro `::cxx::map_rust_error_to_cxx_exception` (in Nightly builds, run with -Z macro-backtrace for more info)
